frontend/packages/pipelines-plugin/src/utils/pipeline-actions.tsx

```diff
@@ -373,15 +458,21 @@ export const getPipelineKebabActions = (
 
 export const getPipelineRunKebabActions = (
   operatorVersion: SemVer,
-  taskRuns: TaskRunKind[],
+  taskRuns?: TaskRunKind[],
   redirectReRun?: boolean,
+  taskRunStatusObj?: TaskStatus,
 ): KebabAction[] => [
   redirectReRun
     ? (model, pipelineRun) => rerunPipelineRunAndRedirect(model, pipelineRun)
     : (model, pipelineRun) => reRunPipelineRun(model, pipelineRun),
-  (model, pipelineRun) => stopPipelineRun(model, pipelineRun, operatorVersion, taskRuns),
-  (model, pipelineRun) => viewPipelineRunSBOM(model, pipelineRun, taskRuns),
-  (model, pipelineRun) => cancelPipelineRunFinally(model, pipelineRun, taskRuns),
+  !taskRunStatusObj
+    ? (model, pipelineRun) => stopPipelineRun(model, pipelineRun, operatorVersion, taskRuns)
+    : (model, pipelineRun) =>
+        stopPipelineRunForTaskRunStatus(model, pipelineRun, operatorVersion, taskRunStatusObj),
+  !taskRunStatusObj
+    ? (model, pipelineRun) => cancelPipelineRunFinally(model, pipelineRun, taskRuns)
+    : (model, pipelineRun) =>
+        cancelPipelineRunFinallyForTaskRunStatus(model, pipelineRun, taskRunStatusObj),
```

Comment:
Instead of maintaining two versions of `Stop`  and `Cancel` actions ( one for `taskrunsStatusObj` and one fore `taskRuns`) , cant we just have a utility function that takes `taskRuns` and convert it to `taskRunStatusObj`, so that we can just keep only the new version of actions, wdyt?

Reply:
I have made two versions because getPipelineRunKebabActions is used by  PipelineRunDetailsPage and RepositoryPipelineRunRow also. So I thought of keeping taskRuns since the other modules are already sending taskRuns. @jerolimov / @vikram-raj any thoughts on this?